### PR TITLE
Castle walls fix

### PR DESCRIPTION
**Pull request by Tarcontar**
The walls of fortress maps where all positioned in the center of the fortress, since the obejct property "GeometryRotationAnchorOffset" was not handled properly.
Also fixed the rotation of fortresses (bfme2 starting buildings) placed at the player start positions.

### CODE REVIEW

File: src/OpenSage.Game/Logic/Object/GameObject.cs [hunk 1]
```diff
@@ -23,12 +23,12 @@ public sealed class GameObject : DisposableBase
         internal static GameObject FromMapObject(
             MapObject mapObject,
             AssetStore assetStore,
-            GameObjectCollection parent,
-            in Vector3 position,
+            GameObjectCollection gameObjects,
```
Comment (feliwir) [inline src/OpenSage.Game/Logic/Object/GameObject.cs:26]: Why did you rename that?

Comment (Tarcontar) [reply src/OpenSage.Game/Logic/Object/GameObject.cs:26]: since 'parent' was very misleading in that context
